### ISSUE TITLE
Focus moves unexpectedly and creates keyboard trap - IE only (critical)

### ISSUE THREAD

**Issue by MelanieHotchkiss** [magda-io/magda#1910]
### Problem description
Intopia issue 39 - When an Internet Explorer user adds search options or refinements – by way of autocomplete suggestions or by filtering their search with facets – the search history creates a keyboard trap. If the user moves focus, focus moves immediately to the top of the page. Tabbing forward goes only as far as the search box with the displayed search history and then focus returns to the top of the page. The keyboard user is trapped.
This only occurs if the main search box has built a search history, or if facets are applied to a search. Otherwise, there is no keyboard trap.
Note that in Firefox, focus also shifts to the top of the page when a user attempts to tab out of a search box with a search history, but the user can then tab through to the search box and past it if they wish – there’s no keyboard trap, but the unexpected focus shift is confusing for users.

### Problem reproduction steps

Ensure that the keyboard user can move focus out of the search box and forward through the page.

### Screenshot / Design / File reference
[39] Focus moves unexpectedly and creates keyboard trap (IE only)
intopia.digi tal 44
Type: WCAG 2.1 failure
Fails 2.1.2 No Keyboard Trap - Level A
Location: Search box, Search page
Platform: Desktop
Severity: Critical
What’s the issue?
When an Internet Explorer user adds search options or refinements – by way of autocomplete suggestions or by filtering their search with facets – the search history creates a keyboard trap. If the user moves focus, focus moves immediately to the top of the page. Tabbing forward goes only as far as the search box with the displayed search history and then focus returns to the top of the page. The keyboard user is trapped.
This only occurs if the main search box has built a search history, or if facets are applied to a search. Otherwise, there is no keyboard trap.
Note that in Firefox, focus also shifts to the top of the page when a user attempts to tab out of a search box with a search history, but the user can then tab through to the search box and past it if they wish – there’s no keyboard trap, but the unexpected focus shift is confusing for users.
Why it matters
The keyboard user’s fundamental ability to navigate the page is broken.
Our recommendation
Ensure that the keyboard user can move focus out of the search box and forward through the page.
